### ISSUE TITLE
Sorting out Nightfall completion stats

### ISSUE THREAD

**Issue by cortical-iv** [Bungie-net/api#635]
Related issue:    
https://github.com/Bungie-net/api/issues/304    

I have a clan leaderboard that includes nightfall completions, extracted using the GetHistoricalStats endpoint. For this question let me just focus on one person just I can try to be less confusing. 

I used to use mode 16 (Nightfall), and before the February change, person X had 122 completions. Their stats didn't increase after the February change, so they were stuck at 122 (though the past week or so it moved up to 125 -- though note between February and now I know this person did a lot more than 3 nightfalls).  

Stepping back, with the February change, I believe modes 46 and 47 were added. I want to start using mode 46 (Scored Nightfall). This same person X has 103 completions returned when I use mode 46.

So my question is, how many nightfalls has this person done? Is it the sum of modes 16 and 46? Going forward, is this what I should use? Has mode 16 started to update again? Do modes 46/47 not include stats from before the February update, and if not, would it be impossible to change that, and pull those stats in so we can start using just one mode to calculate total Nightfalls?

Any help sorting this out would be greatly appreciated. Sorry for the long post. :octopus:

**Comment by vthornheart-bng** [comment]
Ah, yes - so if you want completions across history and regardless of the type of nightfall, you indeed will want to grab the Nightfall (16), HeroicNightfall (17), ScoredNightfall (46), and ScoredHeroicNightfall (47) modes.  Given the history of how Nightfall has ended up with these different variants - variants that have been created and retired across D2's history - I could see us wanting to make a new mode "AllNightfall" that would encompass all of these.  I don't know if/when we'll have time to do that, but I could see that being helpful for such consolidation.

I'll add a task - I can't promise it's something we'll get to anytime soon however, so for now you'll have to aggregate them manually.

**Comment by vthornheart-bng** [comment]
TFS 706853, but again I can't promise we'll be able to get to that anytime soon... so for now, you'll have to aggregate them manually (sum up all the modes).

Whatever mode it was when the PGCR was saved, that's the mode it will always be (except in some rare circumstances where we were able to reprocess known-broken PGCRs, which is pretty rare and often can't be done, and definitely wouldn't have been done for any Nightfalls up to now).

**Comment by vthornheart-bng** [comment]
Yeah, I'd definitely feel better about it if we had a single overarching aggregate mode now that it's spread out so far.  It's one of those things that creeps up on you as time goes by!  I'm glad you mentioned it.
